src/pages/events.tsx

```diff
@@ -0,0 +1,404 @@
+                                                                    <button className="button-event" onClick={() => handleRegisterClick(event.name + event.time)}>Register</button> 
+                                                                )}
+                                                                {registrationStates[event.name + event.time] && (
+                                                                    <div>
+                                                                        <form>
+                                                                            <p>Would you like to register as a participant or volunteer?</p>
+                                                                            <div className="radio">
+                                                                                <label>
+                                                                                    <input type="radio" value="participant" name="registration" /> participant  <br></br>
+                                                                                    <input type="radio" value="volunteer" name="registration" /> volunteer  <br></br>
+                                                                                </label>
+                                                                            </div>
+
+                                                                            <p>Please select the names of the people in your group who will be participating</p>
+                                                                            <div className="checkbox">
+                                                                                <input type="checkbox" value="registrant1" name="group" /> registrant1 
+                                                                                <input type="checkbox" value="registrant2" name="group" /> registrant2
+                                                                                <input type="checkbox" value="registrant3" name="group" /> registrant3
+                                                                            </div>
+                                                                            <div className="button-container">
+                                                                                <button className="submit-button" type="submit">Submit</button>
+                                                                            </div>
+                                                                        </form>
+                                                                    </div>
+                                                                )}
+                                                                <button className="button-event" onClick={() => handleCancelClick(event.name + event.time)}>Cancel</button>
```

Comment:
This does nothing before you click "Register" and it shouldn't appear after you click it.